apps/src/codebridge/hooks/useInitialSources.ts

```diff
@@ -15,23 +19,41 @@ export const useInitialSources = (defaultSources: ProjectSources) => {
   const levelStartSource = useAppSelector(
     state => state.lab.levelProperties?.source
   );
+  const exemplarSources = useAppSelector(
+    state => state.lab.levelProperties?.exemplarSources
+  );
   // We memoize this object so that it doesn't cause an unexpected re-render.
   const projectStartSource: ProjectSources | undefined = useMemo(
     () => (levelStartSource ? {source: levelStartSource} : undefined),
     [levelStartSource]
   );
   const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
+  const isEditingExemplar = getAppOptionsEditingExemplar();
+  const isViewingExemplar = getAppOptionsViewingExemplar();
```

Comment:
How expensive is it to `getScriptData('appoptions')`? We have six utils now that get these options and then return one particular value (`channel`, `editBlocks`,  `isEditingExemplar`, `isViewingExemplar`, `levelId`, `share`). 

I'm wondering if it would be more efficient in any way to have a single util that returns an object with all of the values we potentially need than to be repeatedly parsing JSON data from a script element in the DOM.

Reply:
I think it's very cheap, it's a lookup followed by a json parse of a pretty small string.